app/javascript/selectors/answersSelector.js

```diff
@@ -1,10 +1,44 @@
 import {
   getAllActiveQuestions,
   getAllActiveQuestionsForCategory,
-  getAllQuestionsByCategoryId
+  getAllActiveQuestionIdsForCategory,
 } from "./questionsSelector";
 import { getCategoriesForBuilding } from "./categoriesSelector";
 
+export function getAnswersForBuilding(buildingId, state) {
+  let answers = state.buildings[buildingId].answers;
+  answers = Object.keys(answers).filter((id) => {
+    let email = answers[id].delegation_email;
+    let text = answers[id].text;
+    return !email && !text;
+  })
+  .reduce((newAnswers, id) => {
+    newAnswers[id] = answers[id];
+    return newAnswers
+  }, {});
+  return answers
+}
+
+export function getAnswersForCategoryAndBuilding(categoryId, buildingId, state) {
+  const buildingQuestions = state.buildings[buildingId].questions;
+  let categoryQuestions = state.categories[categoryId].questions;
+  categoryQuestions = buildingQuestions.filter((id) => {
+    return categoryQuestions.includes(Number(id));
+  })
+  
+  let answers = state.buildings[buildingId].answers;
+  answers = Object.keys(answers).filter((id) => {
+    let email = answers[id].delegation_email;
+    let text = answers[id].text;
+    return !email && !text && categoryQuestions.includes(id);
+  })
+  .reduce((newAnswers, id) => {
+    newAnswers[id] = answers[id];
+    return newAnswers
+  }, {});
+  return answers;
```

Comment:
I don't think `.reduce` is doing anything here, since you're just returning `answers` (the array of IDs, rather than a reconstructed object). It seems the store is being updated correctly so you can probably remove this part!

Reply:
I believe am actually returning the reconstructed object here, the filter function was just spaced weirdly but answers the result of the filter and the reduce.

Reply:
OH oops I see now gotcha 👍